list/list.js

```diff
@@ -1004,8 +1005,8 @@ assign(DefineList.prototype, {
 	map: function(callback, thisArg) {
 		var mappedList = [],
 			self = this;
-		this.each(function(item, index, list) {
-			var mapped = callback.call(thisArg | self, item, index, self);
+		canReflect.eachIndex(this, function(item, index, list) {
+			var mapped = canReflect.call(callback, thisArg | self, item, index, self);
```

Comment:
I think we might be able to get away with `callback.call`.  I have to think about if we need to support arbitrary function like things here and now.